src/memoize.js

```diff
@@ -10,16 +10,35 @@ import type {Context} from './types.js';
 
 type MemoizeFn<A> = (ctx: Context) => A;
 
-function Container() {}
-
 export function memoize<A>(fn: MemoizeFn<A>): MemoizeFn<A> {
-  const memoizeKey = __NODE__ ? Symbol('memoize-key') : new Container();
-  return function memoized(ctx: Context) {
-    if (ctx.memoized.has(memoizeKey)) {
-      return ctx.memoized.get(memoizeKey);
+  if (__BROWSER__) {
+    return browserMemoize(fn);
+  }
+
+  const wm = new WeakMap();
```

Comment:
We could put this logic be in an else block

Reply:
Actually, we ship core-js polyfills when necessary, so perhaps we can just assume WeakMap exists.

Reply:
I'm not sure but it the core-js polyfill seems to use this weak collection implementation: https://github.com/zloirock/core-js/blob/master/packages/core-js/internals/collection-weak.js